src/instruction/var.rs

```diff
@@ -116,6 +121,18 @@ impl Instruction for Var {
     }
 }
 
+impl TypeCheck for Var {
+    type_cache!(ty);
+
+    fn resolve_type(&self, ctx: &mut Context) -> CheckedType {
+        match self.instance.ty() {
+            None => CheckedType::Void,
```

Comment:
I probably understand it wrong, but does that mean "if variable doesn't exist return void", shouldn't this be an impossible state or an error ?

Reply:
This might need to be an error case indeed. The `ObjectInstance` struct keeps its type, so there should be some refactoring done to match it